### ISSUE TITLE
Feature request: explicitly-defined padding in ConvTranspose

### ISSUE THREAD

**Issue by calvinmccarter-at-lightmatter** [onnx/tensorflow-onnx#1501]
Currently [(here in onnx_opset/nn.py)](https://github.com/onnx/tensorflow-onnx/blob/9c7b56c7321927666eb2c00d15e6db16a1b7de1f/tf2onnx/onnx_opset/nn.py#L440), whenever possible a ConvTranspose node is created using `output_shape`-based padding (which is dynamic) rather than explicitly-defined padding.  In my opinion, it would be preferable to set `output_shape=None` and instead compute the corresponding `pads` from this `output_shape`. It's a lot more straightforward to convert ONNX models to PyTorch models when they use `pads` rather than `output_shape`, since `pads` is only computable if you also have an input datum (as one does with ONNX). And in my opinion, it's also more intuitive when examining the resulting ONNX model (in say, Netron).

Alternatively, the current condition `if output_shape is not None` could be replaced with something like `if FLAG_NAME and output_shape is not None`.

Any thoughts before I start working on a PR?

**Comment by TomWildenhain-Microsoft** [comment]
Sure, using pads instead of output shape when possible is fine.  If you make a PR, I'll review and merge it.

**Comment by calvinmccarter-at-lightmatter** [comment]
@TomWildenhain-Microsoft -- just made a PR, thanks!
